### DISCUSSION TITLE
How to handle value objects in forms?

### DISCUSSION THREAD

**Discussion by panlatent** [filamentphp/filament#14390] [General]
Filament does not seem to have out-of-the-box support for value objects.

```php
public static function form(Form $form): Form
{
 return $form
            ->schema([
                    Forms\Components\Grid::make()
                            ->relationship('database')
                            ->schema([
                                     Forms\Components\TextInput::make('access.hostname')
                            ]);
            ]);
}
```

By defining Casts on the Model and adding Wireable. At this point, the value of the input box is blank, but the value can be set correctly. I found that I can get the correct value in the `afterStateHydrated()` method.

Any suggestions? I'm new to Filament and Livewire and just got back to Laravel.

**Comment by Urbanproof** [comment]
I'm probably late in to the party, but just in case someone else stumbles into this; Filament is built on top of Livewire (among other things), so using [Livewire synthesizers](https://livewire.laravel.com/docs/synthesizers) should get you there.

**Reply by jonjakoblich** [reply]
Stumbling upon this much later, but this resolved the issue for me. Thank you!
